module/zfs/zfs_fm.c

```diff
@@ -1444,6 +1444,48 @@ zfs_ereport_fini(void)
 	mutex_destroy(&recent_events_lock);
 }
 
+void
+zfs_ereport_snapshot_post(const char *subclass, spa_t *spa, const char *name)
+{
+	nvlist_t *aux;
+
+	aux = fm_nvlist_create(NULL);
+	nvlist_add_string(aux, "snapshot_name", name);
+
+	zfs_post_common(spa, NULL, FM_RSRC_CLASS, subclass, aux);
+	fm_nvlist_destroy(aux, FM_NVA_FREE);
+}
+
+void
+zfs_ereport_zvol_post(const char *subclass, const char *name, const char *bsd,
+    const char *rbsd)
+{
+	nvlist_t *aux;
+	char *r;
+	spa_t *spa;
+	boolean_t has_lock = B_FALSE;
+
+	has_lock = mutex_owned(&spa_namespace_lock);
+	if (!has_lock) mutex_enter(&spa_namespace_lock);
```

Comment:
Is it possible for `spa_namespace_lock` be acqured between these two lines?:
```C
	has_lock = mutex_owned(&spa_namespace_lock);
￼	if (!has_lock) mutex_enter(&spa_namespace_lock);
```

Reply:
For us it was a test to see if `this thread` was already holding the lock. Either we already own it, or we need to grab it. There should be no race, I don't think.

Reply:
Got it, thanks